sdks/aperture-go/sdk/flow.go

```diff
@@ -161,6 +168,10 @@ func (f *flow) DeleteResultCache(ctx context.Context, opts ...grpc.CallOption) K
 		return newKeyDeleteResponse(ErrResultCacheKeyNotSet)
 	}
 
+	if f.checkResponse == nil {
+		return newKeyDeleteResponse(ErrAgentDisconnected)
```

Comment:
Should return flow.Errora() instead which captures the precise reason: https://github.com/fluxninja/aperture/blob/bccf86637baf91c060d6945ecfce5bd1b947cf35/sdks/aperture-go/sdk/client.go#L176